### PR TITLE
Fixed docker based development issue.

### PR DESCRIPTION
**Pull request by k2glyph**
- Create example config and added config.yml and dbconf.yml to gitignore
- Updated docker-compose.yml and remove unwanted docker-compose.db.yaml and dockerfile.dev
- Update Readme.md and Makefile
- Using MariaDB for database

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,4 +13,6 @@ learning/
 coverage.out
 ./out
 .scannerwork/
-meroedu
\ No newline at end of file
+meroedu
+config.yml
+migrator/dbconf.yml
```
Comment (trangmaiq) [inline .gitignore:18]: Please fix EOL here.

File: scripts/wait-for-db.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+until docker container exec -it db mariadb-admin ping -P 3306 -uroot -proot | grep "mysqld is alive" ; do
+  >&2 echo "Database is unavailable - waiting for it... 😴"
+  sleep 1
+done
```
Comment (trangmaiq) [inline scripts/wait-for-db.sh:5]: Please fix missing EOL here.

File: docker-compose.yaml [hunk 3]
```diff
@@ -4,32 +4,37 @@ services:
...
-    image: adminer
-    restart: always
-    ports:
-      - 8080:8080
\ No newline at end of file
+      retries: 6
+
+volumes:
+  db:
+    driver: local
```
Comment (trangmaiq) [inline docker-compose.yaml:40]: Please fix the missing EOL here also.

File: config.example.yml [hunk 4]
```diff
@@ -7,7 +7,7 @@ context:
 database:
   params:
     parseTime: "true"
-  host: localhost
+  host: db
```
Comment (trangmaiq) [inline config.example.yml:10]: Not quite sure why we have this change?

File: .gitignore (hunk 1 above)
Comment (k2glyph) [reply .gitignore:18]: Fixing.

File: config.example.yml (hunk 4 above)
Comment (k2glyph) [reply config.example.yml:10]: This is the container name of database

**Review by trangmaiq — APPROVED** [review]
Thanks for fixing it.

**Pull request merged**